src/Symfony/Component/Serializer/Normalizer/DataUriNormalizer.php

```diff
@@ -81,12 +81,12 @@ public function supportsNormalization($data, $format = null)
      * @see https://gist.github.com/bgrins/6194623
      *
      * @throws InvalidArgumentException
-     * @throws UnexpectedValueException
+     * @throws NotNormalizableValueException
      */
     public function denormalize($data, $class, $format = null, array $context = array())
     {
         if (!preg_match('/^data:([a-z0-9][a-z0-9\!\#\$\&\-\^\_\+\.]{0,126}\/[a-z0-9][a-z0-9\!\#\$\&\-\^\_\+\.]{0,126}(;[a-z0-9\-]+\=[a-z0-9\-]+)?)?(;base64)?,[a-z0-9\!\$\&\\\'\,\(\)\*\+\,\;\=\-\.\_\~\:\@\/\?\%\s]*\s*$/i', $data)) {
-            throw new UnexpectedValueException('The provided "data:" URI is not valid.');
+            throw new NotNormalizableValueException('The provided "data:" URI is not valid.');
```

Comment:
IMHO this example here should actually continue to throw the old exception. This is decodeable, but just not valid in the domain. I guess this example would be a 422 instead?